src/quantity-selector/index.ts

```diff
@@ -0,0 +1,68 @@
+import { alias, tag, Events, Selectors, Tag } from '@storefront/core';
+
+const INITIAL_INDEX = 0;
+
+@alias('selector')
+@tag('gb-quantity-selector', require('./index.html'))
+class QuantitySelector {
+
+  state: QuantitySelector.State = {
+    selected: INITIAL_INDEX,
+    options: []
+  };
+
+  init() {
+    this.getOptions(this.props.quantity);
+
+    // args for this event is a url link
+    this.flux.on(Events.URL_UPDATED, this.reset);
+  }
+
+  onMount() {
+    this.flux.emit('selector:update', this.state.options[this.state.selected].value);
+  }
+
+  getOptions = (quantity: number) => {
+    let options = [];
```

Comment:
const options

Reply:
but options will change afterwards

Reply:
The keys of options will change but option itself will be the same object.